### PR TITLE
Add a patch to pin openblas <0.3.26 for older scipy builds

### PR DESCRIPTION
**Pull request by hmaarrfk**
xref: https://github.com/conda-forge/scipy-feedstock/issues/266#issuecomment-1907148347

cc: @h-vetinari 

diff: https://gist.github.com/hmaarrfk/b69c7844166efc7cd61f98ae2a4548d5

Checklist

* [ ] Used a static YAML file for the patch if possible ([instructions](https://github.com/conda-forge/conda-forge-repodata-patches-feedstock/blob/main/recipe/README.md)).
* [ ] Only wrote code directly into `generate_patch_json.py` if absolutely necessary.
* [ ] Ran `pre-commit run -a` and ensured all files pass the linting checks.
* [ ] Ran `python show_diff.py` and posted the output as part of the PR.
* [ ] Modifications won't affect packages built in the future.

### CODE REVIEW

**Comment by h-vetinari** [comment]
I think the approach is correct, but this is touching _a lot_ of builds (even though the addition of a constraint for a package that didn't exist at the time these builds were published should be pretty harmless). Not sure what the best way to handle this is. CC @conda-forge/core.

File: recipe/patch_yaml/scipy.yaml [hunk 1]
```diff
@@ -0,0 +1,26 @@
+# 2024/01 -- hmaarrfk
+# scipy version 1.20.0 build 2 was patched
```
Comment (h-vetinari) [inline recipe/patch_yaml/scipy.yaml:2]:
```suggestion
# scipy version 1.12.0 build 2 was patched
```

File: recipe/patch_yaml/scipy.yaml [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  version: 1.12.0
+  build_number_lt: 2
+
+then:
+  - add_constrains: libopenblas <0.3.26
+
+---
+
+# 2024/01 -- hmaarrfk
+# scipy version 1.20.0 build 2 was patched
```
Comment (h-vetinari) [inline recipe/patch_yaml/scipy.yaml:17]:
```suggestion
# scipy version 1.12.0 build 2 was patched
```

**Review by h-vetinari — APPROVED** [review]
LGTM aside from some comment nits.

File: recipe/patch_yaml/scipy.yaml (hunk 2 above)
Comment (hmaarrfk) [reply recipe/patch_yaml/scipy.yaml]: thx!

**Comment by h-vetinari** [comment]
Any comments? This is hitting scipy installs quite widely so I think we should do it (e.g. someone who's not able to update to 1.12, which is the only version that's fixed ATM)

**Comment by beckermr** [comment]
merge away!

**Pull request merged**